features/attached_commands.feature

```diff
@@ -0,0 +1,85 @@
+@uses.config.contract_token
+Feature: Command behaviour when attached to an UA subscription
+
+    Scenario: Attached refresh in a trusty lxd container
+        Given a trusty lxd container with ubuntu-advantage-tools installed
+        When I attach contract_token with sudo
+        And I run `ua refresh` as non-root
+        Then I will see the following on stderr:
+            """
+            This command must be run as root (try using sudo)
+            """
+            #When The machine is attached
+        When I run `ua refresh` with sudo
+        Then I will see the following on stdout:
+            """
+            Successfully refreshed your subscription
+            """
+
+    Scenario: Attached disable of a known service in a trusty lxd container
+        Given a trusty lxd container with ubuntu-advantage-tools installed
+        When I attach contract_token with sudo
+        And I run `ua disable livepatch` as non-root
+        Then I will see the following on stderr:
+            """
+            This command must be run as root (try using sudo)
+            """
+        When I run `ua disable livepatch` with sudo
+        Then I will see the following on stdout:
+            """
+            Livepatch is not currently enabled
+            See: sudo ua status
+            """
+
+    Scenario: Attached disable of an unknown service in a trusty lxd container
+        Given a trusty lxd container with ubuntu-advantage-tools installed
+        When I attach contract_token with sudo
+        And I run `ua disable foobar` as non-root
+        Then I will see the following on stderr:
+            """
+            This command must be run as root (try using sudo)
+            """
+        When I run `ua disable foobar` with sudo
+        Then I will see the following on stderr:
+            """
+            Cannot disable 'foobar'
+            For a list of services see: sudo ua status
+            """
+
+    Scenario: Attached disable of a known enabled service in a trusty lxd container
+        Given a trusty lxd container with ubuntu-advantage-tools installed
+        When I attach contract_token with sudo
+        And I run `ua disable esm-infra` as non-root
+        Then I will see the following on stderr:
+            """
+            This command must be run as root (try using sudo)
+            """
+        When I run `ua disable esm-infra` with sudo
+        Then I will see the following on stdout:
+            """
+            Updating package lists
+            """
+        When I run `ua status` with sudo
+        Then stdout matches regexp:
+            """
+            esm-infra    +yes      +disabled +UA Infra: Extended Security Maintenance
+            """
+
```

Comment:
I would be nice to also add a step that allows us to validate that esm repositories are properly disabled on trusty to the magic -32768 because this behavior will differ on Trusty (which advertizes available esm packages) versus Xenial and Bionic (which should not have any esm repos seen when esm is disabled).

```suggestion
    When I run `apt-cache policy` with sudo
    Then stdout matches regexp
        """
        -32768 https://esm.ubuntu.com/ubuntu/ trusty-infra-updates/main amd64 Packages
          +origin esm.ubuntu.com
        -32768 https://esm.ubuntu.com/ubuntu/ trusty-infra-security/main amd64 Packages
          +origin esm.ubuntu.com
        """
```

Reply:
hi @patriciadomin your suggestion failure that you pinged about per https://pastebin.canonical.com/p/RMz2PM9Xcr/
is because you ran `apt-cache policy ubuntu-advantage-tools` which only tells you about the specific package.

Note the above command I mentioned was only `apt-cache policy`. Which gives you a list of all apt repositories that the machine has access to. this will give you the expected output.